server/engine/src/main/java/org/eclipse/lsp/cobol/service/copybooks/CopybookServiceImpl.java

```diff
@@ -124,11 +135,32 @@ public CopybookModel resolve(
     }
   }
 
+  private void storeCopybookUsageReference(CopybookName copybookName, String documentUri, CopybookModel copybookModel) {
```

Comment:
This logic is not needed in the `CopybookService`, let's move it to separate reference repository, for example `ReferenceRepo`